File: 49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> map = new HashMap<>();
        for(String str : strs){
            char[] chars = str.toCharArray();
            Arrays.sort(chars);
            String str_s = String.valueOf(chars);
            if(!map.containsKey(str_s)){
                map.put(str_s, new ArrayList<>());
            }
            
            map.get(str_s).add(str);
        }
        
        List<List<String>> result = new ArrayList<>();
        for(List<String> set : map.values()){
            result.add(set);
        }
        
        return result;
    }
}

File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        hashmap = {}
        for str in strs:
            currChars = list(str)
            currChars.sort()
            reconstructedStr = ''.join(currChars)
            if reconstructedStr not in hashmap:
                hashmap[reconstructedStr] = []
            hashmap[reconstructedStr].append(str)
        
        ans = []
        for key, value in hashmap.items():
            ans.append(value)
        
        return ans